### PR TITLE
useQuery: adjust for rules of React linting

### PR DESCRIPTION
**Pull request by phryneas**
Another one for #11511...

### CODE REVIEW

File: .size-limits.json [hunk 1]
```diff
@@ -1,4 +1,4 @@
 {
-  "dist/apollo-client.min.cjs": 39573,
+  "dist/apollo-client.min.cjs": 39607,
```
Comment (phryneas) [inline .size-limits.json:2]: This PR actually doesn't change any behaviour, it just tries to give the lint rule some way of linting this hook. I'm not happy that we get this kind of size increase for that.

File: src/react/hooks/useQuery.ts [hunk 2]
```diff
@@ -526,6 +551,8 @@ class InternalState<TData, TVariables extends OperationVariables> {
       this.observable || // Reuse this.observable if possible (and not SSR)
       this.client.watchQuery(this.getObsQueryOptions()));
 
+    // React Hook "React.useMemo" cannot be called in a class component.
+    // eslint-disable-next-line react-hooks/rules-of-hooks
```
Comment (phryneas) [inline src/react/hooks/useQuery.ts:555]: Ignore the linter error here for now - it goes away once we actually pull that rule in.

File: src/react/hooks/useQuery.ts [hunk 3]
```diff
@@ -215,104 +215,129 @@ class InternalState<TData, TVariables extends OperationVariables> {
...
+
+      const disableNetworkFetches = internalState.client.disableNetworkFetches;
+
+      const result = useSyncExternalStore(
+        React.useCallback(
+          (handleStoreChange) => {
+            // there is nothing to really do here with `disableNetworkFetches`,
+            // we need to reference it here to ensure the effect depends on it
+            // and resubscribes when it changes and the component is re-rendered by chance
+            disableNetworkFetches;
```
Comment (jerelmiller) [inline src/react/hooks/useQuery.ts:254]: What an odd one! Would love a rewrite of this hook to try and remove this oddity 😆

**Review by jerelmiller — APPROVED** [review]
Creative way to get this linted. Thanks!

**Pull request closed without merging**